src/main/java/com/example/springboot/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+package com.example.springboot.service.impl;
+
+import com.example.springboot.dto.order.CreateOrderRequestDto;
+import com.example.springboot.dto.order.OrderResponseDto;
+import com.example.springboot.dto.order.UpdateOrderRequestDto;
+import com.example.springboot.dto.orderitem.OrderItemDto;
+import com.example.springboot.mapper.OrderItemMapper;
+import com.example.springboot.mapper.OrderMapper;
+import com.example.springboot.model.Order;
+import com.example.springboot.model.OrderItem;
+import com.example.springboot.model.User;
+import com.example.springboot.repository.order.OrderRepository;
+import com.example.springboot.repository.orderitem.OrderItemRepository;
+import com.example.springboot.repository.user.UserRepository;
+import com.example.springboot.service.OrderService;
+import com.example.springboot.service.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private OrderRepository orderRepository;
+    private OrderItemRepository orderItemRepository;
+    private UserRepository userRepository;
+    private ShoppingCartService shoppingCartService;
+    private OrderItemMapper orderItemMapper;
+    private OrderMapper orderMapper;
+
+    @Transactional
+    @Override
+    public OrderResponseDto saveOrder(Long userId, CreateOrderRequestDto orderRequestDto) {
+        User user = userRepository.findById(userId).orElseThrow(()
+                        -> new RuntimeException("Can't find user by id " + userId));
+        Order order = new Order();
+        order.setUser(user);
+        Set<OrderItem> orders = shoppingCartService.getShoppingCartByUserId(userId)
+                .getCartItems()
+                .stream()
+                .map(orderItemMapper::toEntity)
+                .collect(Collectors.toSet());
+        order.setOrderItems(getOrderItemsByUserId(userId));
+        order.setStatus(Order.Status.PENDING);
+        order.setOrderDate(LocalDateTime.now());
+        return null;
```

Comment:
never return null. 

make this method void in case you need to return nothing.